fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -1130,21 +1122,21 @@ struct RestoreClusterImpl {
 		return Void();
 	}
 
-	ACTOR static Future<Void> markManagementTenantAsError(RestoreClusterImpl* self,
-	                                                      Reference<typename DB::TransactionT> tr) {
-		state Optional<TenantMapEntry> tenantEntry = wait(tryGetTenantTransaction(tr, self->tenantName));
+	ACTOR static Future<Void> markManagementTenantAsError(Reference<typename DB::TransactionT> tr,
+	                                                      TenantName tenantName) {
+		state Optional<TenantMapEntry> tenantEntry = wait(tryGetTenantTransaction(tr, tenantName));
 
 		if (!tenantEntry.present()) {
 			return Void();
 		}
 
 		tenantEntry.get().tenantState = TenantState::ERROR;
-		ManagementClusterMetadata::tenantMetadata().tenantMap.set(tr, self->tenantName, tenantEntry.get());
+		ManagementClusterMetadata::tenantMetadata().tenantMap.set(tr, tenantName, tenantEntry.get());
```

Comment:
I added an error string field that you can also set on the tenant to indicate the details of the error, which might be useful in this case.